tests/utilities/nidmm_measurement/__init__.py

```diff
@@ -0,0 +1,87 @@
+import math
+import pathlib
+from typing import Iterable, Sequence, Tuple
+
+import nidmm
+
+import ni_measurementlink_service as nims
+
+
+service_directory = pathlib.Path(__file__).resolve().parent
+measurement_service = nims.MeasurementService(
+    service_config_path=service_directory / "NIDMMMeasurement.serviceconfig",
+    version="0.1.0.0",
+    ui_file_paths=[
+        service_directory,
+    ],
+)
+
+
+@measurement_service.register_measurement
+@measurement_service.configuration("pin_names", nims.DataType.PinArray1D, ["Pin1"])
+@measurement_service.configuration("multi_session", nims.DataType.Boolean, False)
+@measurement_service.output("session_names", nims.DataType.StringArray1D)
+@measurement_service.output("resource_names", nims.DataType.StringArray1D)
+@measurement_service.output("channel_lists", nims.DataType.StringArray1D)
+@measurement_service.output("connected_channels", nims.DataType.StringArray1D)
+@measurement_service.output("signals_out_of_range", nims.DataType.BooleanArray1D)
+@measurement_service.output("absolute_resolutions", nims.DataType.DoubleArray1D)
+def measure(
+    pin_names: Iterable[str],
+    multi_session: bool,
+) -> Tuple[
+    Iterable[str], Iterable[str], Iterable[str], Iterable[str], Iterable[bool], Iterable[float]
+]:
+    """NI-DMM MeasurementLink test service."""
+    if multi_session:
+        with measurement_service.context.reserve_sessions(pin_names) as reservation:
+            with reservation.initialize_nidmm_sessions() as session_infos:
+                connections = reservation.get_nidmm_connections(pin_names)
+                assert all([session_info.session is not None for session_info in session_infos])
+                signals_out_of_range, absolute_resolutions = _get_dmm_readings(session_infos)
+
+                return (
+                    [session_info.session_name for session_info in session_infos],
+                    [session_info.resource_name for session_info in session_infos],
+                    [session_info.channel_list for session_info in session_infos],
+                    [connection.channel_name for connection in connections],
+                    signals_out_of_range,
+                    absolute_resolutions,
+                )
+
+    else:
+        with measurement_service.context.reserve_session(pin_names) as reservation:
+            with reservation.initialize_nidmm_session() as session_info:
+                connection = reservation.get_nidmm_connection(list(pin_names)[0])
+                assert session_info.session is not None
+                signals_out_of_range, absolute_resolutions = _get_dmm_readings([session_info])
+
+                return (
+                    [session_info.resource_name],
+                    [session_info.resource_name],
+                    [session_info.channel_list],
+                    [connection.channel_name],
+                    [list(signals_out_of_range)[0]],
+                    [list(absolute_resolutions)[0]],
```

Comment:
Type hint these as Sequence or List instead of Iterable so you don't need to build a temp list.

I guess you could do that for pin_names too...